### DISCUSSION TITLE
Cross-Version Runtime Guarantee by Wire?

### DISCUSSION THREAD

**Discussion by prathamk** [square/wire#3055] [Q&A]
Hello,

We were looking to decode GPB encoded messages, and evaluating the Google packages (protoc and protobuff-java). Those were working fine, though the generated code is heavy from our need perspective. But we ran into this problem, that they don't support gencode to run on newer runtime across major releases (We kind of have this requirement).

Does wire support Cross-Version runtime as it upgrades internal dependencies (as I see protoc, and protobuff-java is also mentioned in dependencies)? Or is there a same requirement here also, that across releases code needs to be re-generated?
When and in which scenarios is code re-generation needed by wire?

Thanks//Pratham

**Comment by oldergod** [comment]
We've been diligent in allowing generated code to work with newer versions of Wire. If we discovered that we did indeed break something, we would fix it if possible. I would not say however that this is guaranteed.

**Reply by prathamk** [reply]
Thanks @oldergod for your answer.
I have one more related query, I see wire as an alternative to the standard protobuf library where it generates its own code from .proto file and uses them along with wire runtime to decode the data. How is it internally using the protobuf-java, and protoc dependency?

Also is there any benchmarking study already done against the standard protobuf library for deserialization of data. If yes, can you please point us to it.

Thanks//Pratham

**Reply by oldergod** [reply]
There is no dependency on protobuf-java on protoc. Everything is done in house: parsing, generation, serialisation for binary and JSON.

We have a benchmark module here https://github.com/square/wire/tree/master/wire-benchmarks that we ran at one point but it's not automated.

**Reply by prathamk** [reply]
Thanks @oldergod for your answer. Will look into it.
